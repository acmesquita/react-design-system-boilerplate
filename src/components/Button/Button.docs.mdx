import { Meta, Story, Preview } from '@storybook/addon-docs/blocks'
import { Button } from './Button'

<Meta title="Button" component={Button} />

# Button

With `MDX` we can define a story for `Button` right in the middle of our
markdown documentation.

<Preview>
  <Story name="all buttons">
    <Button variant="primary">Primary</Button>
    <Button variant="secondary">Primary</Button>
  </Story>
</Preview>
