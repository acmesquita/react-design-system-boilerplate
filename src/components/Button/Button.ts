import styled from 'styled-components'
import { space, SpaceProps } from 'styled-system'

export const Button = styled.button<SpaceProps>`
  background-color: #fff;
  border: 1px solid #ddd;
  font-weight: bold;
  font-size: 20px;
  border-radius: 0.5em;
  padding: 0.3em 1em;
  ${space};
`
