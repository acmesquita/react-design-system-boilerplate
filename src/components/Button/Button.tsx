/** @jsx jsx */
import { jsx } from 'theme-ui'
import { Button as TUIButton } from '@theme-ui/components'

export const Button = props => <TUIButton {...props} />
