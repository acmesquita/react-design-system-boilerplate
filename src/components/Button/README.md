# Button

A simple button component

## Import

```jsx
import { Button } from 'your-design-system'
// or
import { Button } from 'your-design-system/button'
```

## Usage

```jsx
<Button>Click me please</Button>
```
