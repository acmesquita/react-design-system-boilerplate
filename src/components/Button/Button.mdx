import { Meta, Story, Preview } from '@storybook/addon-docs/blocks'
import { Button } from './Button'

# Button

With `MDX` we can define a story for `Button` right in the middle of our
markdown documentation.

<Story id="primary" />
<Button variant="primary">Primary</Button>
