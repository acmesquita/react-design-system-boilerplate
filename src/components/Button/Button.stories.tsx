import React from 'react'
import { action } from '@storybook/addon-actions'
import { select } from '@storybook/addon-knobs'
import { Button } from './Button'
// import mdx from './Button.mdx'

const label = 'Variant'
const options = {
  Primary: 'primary',
  Secondary: 'secondary',
  Muted: 'muted',
}

export default {
  title: 'Components|Button',
  component: Button,
  parameters: {
    componentSubtitle: 'Handy status label',
    // docs: {
    //   page: mdx,
    // },
  },
}

/**
 * Primary button
 */
export const primary = () => (
  <Button
    onClick={action('clicked')}
    variant={select(label, options, 'primary')}
  >
    Primary
  </Button>
)

/**
 * Secondary button
 */
export const secondary = () => (
  <Button
    onClick={action('clicked')}
    variant={select(label, options, 'secondary')}
  >
    Secondary
  </Button>
)

/**
 * Spacey button
 * Has more  space than the rest
 */
export const spacey = () => (
  <Button
    onClick={action('clicked')}
    variant={select(label, options, 'primary')}
    m={5}
  >
    Secondary
  </Button>
)
