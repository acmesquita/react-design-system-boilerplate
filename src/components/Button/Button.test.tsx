import React from 'react'
import { render } from 'react-testing-library'
import { Button } from './Button'

describe('Given a Button component', () => {
  test('should render a Button', () => {
    const { container } = render(<Button>Hello</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
