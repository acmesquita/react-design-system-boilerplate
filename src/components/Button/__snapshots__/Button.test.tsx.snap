// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component should render a Button 1`] = `
.c0 {
  background-color: #fff;
  border: 1px solid #ddd;
  font-weight: bold;
  font-size: 20px;
  border-radius: 0.5em;
  padding: 0.3em 1em;
}

<button
  class="c0"
>
  Hello
</button>
`;
